test/mlir/conversion/onnx_to_tosa/NN/QuantizeLinear.mlir

```diff
@@ -0,0 +1,17 @@
+// RUN: onnx-mlir-opt --shape-inference --convert-onnx-to-tosa -cse %s -split-input-file | FileCheck %s
+
+/// Test the default behavior of AveragePool with no padding
+func.func @test_quantizeLinear(%arg0 : tensor<32x3x224x224xf32>) -> tensor<32x3x224x224xi8> {
+  %0 = onnx.Constant dense<3.125000e-02> : tensor<f32>                       
+  %1 = onnx.Constant dense<0> : tensor<i8>                                   
+  %2 = "onnx.QuantizeLinear"(%arg0, %0, %1) {axis = 1 : si64} : (tensor<32x3x224x224xf32>, tensor<f32>, tensor<i8>) -> tensor<32x3x224x224xi8>
+  "func.return"(%2) : (tensor<32x3x224x224xi8>) -> ()
+}
+// CHECK-LABEL:  @test_quantizeLinear(%arg0: tensor<32x3x224x224xf32>)
+// CHECK-DAG:    %[[SCALE:.*]] = "tosa.const"() {value = dense<3.125000e-02> : tensor<f32>}
+// CHECK-DAG:    %[[ZP:.*]] = "tosa.const"() {value = dense<0> : tensor<i8>} 
+// CHECK-DAG:    %[[REC:.*]] = "tosa.reciprocal"(%arg0) : (tensor<32x3x224x224xf32>)
+// CHECK-DAG:    %[[MUL:.*]] = "tosa.mul"(%[[REC]], %[[SCALE]]) {shift = 0 : i32} : (tensor<32x3x224x224xf32>, tensor<f32>)
+// CHECK-DAG:    %[[ADD:.*]] = "tosa.add"(%[[MUL]], %[[ZP]]) : (tensor<32x3x224x224xf32>, tensor<i8>)
```

Comment:
I'm kinda surprised to see that, as there doesn't seem to be any assert or error triggering when lowering and executing that.
However this is an issue, should I create a constant tensor of the size of the input then? I was looking into it at first, but I've had trouble figuring out how to retrieve the value from the `zero_point` in onnx as it is an input and not an attribute.

Reply:
You can either get the value from the `tosa.const` https://github.com/onnx/onnx-mlir/blob/375008b5b731bc47894bb4c5715951ef5e915570/src/Conversion/ONNXToTOSA/Math/ReduceMean.cpp#L60-L66 (because of the modules pass) or via the adaptor from the `onnx.const `(there is a helper function for this, too, I think). But I think you are right with creating a new const op